### PR TITLE
Update diagnostic docs for SYSLIB0009

### PR DESCRIPTION
**Pull request by deeprobin**
## Summary

Follow-up for https://github.com/dotnet/runtime/pull/93171.
Change obsoletion documentation: Whole class marked as obsolete. 

See https://github.com/dotnet/runtime/issues/77459.

/cc @MihaZupan 

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/compatibility/9.0.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/core/compatibility/9.0.md) | [Breaking changes in .NET 9](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/9.0?branch=pr-en-us-37452) |
| [docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md) | [API obsoletions with non-default diagnostic IDs (.NET 9)](https://review.learn.microsoft.com/en-us/dotnet/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics?branch=pr-en-us-37452) |
| [docs/fundamentals/syslib-diagnostics/obsoletions-overview.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/fundamentals/syslib-diagnostics/obsoletions-overview.md) | [Obsolete features in .NET 5+](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/obsoletions-overview?branch=pr-en-us-37452) |
| [docs/fundamentals/syslib-diagnostics/syslib0009.md](https://github.com/dotnet/docs/blob/76a9165bdfd7d5c71aada1b2911b5038866861e5/docs/fundamentals/syslib-diagnostics/syslib0009.md) | [SYSLIB0009: AuthenticationManager is not supported](https://review.learn.microsoft.com/en-us/dotnet/fundamentals/syslib-diagnostics/syslib0009?branch=pr-en-us-37452) |

### CODE REVIEW

**Review by MihaZupan — APPROVED** [review]
Thank you

**Comment by gewarren** [comment]
Thank you. Since this is for .NET 9, we shouldn't merge it yet.

**Comment by gewarren** [comment]
We also need to:

- Update https://github.com/dotnet/docs/blob/main/docs/fundamentals/syslib-diagnostics/obsoletions-overview.md
- Create a new article like this but for .NET 9: https://github.com/dotnet/docs/blob/main/docs/core/compatibility/core-libraries/8.0/obsolete-apis-with-custom-diagnostics.md

File: docs/fundamentals/syslib-diagnostics/syslib0009.md [hunk 1]
```diff
@@ -4,13 +4,16 @@ description: Learn about the obsoletions that generate compile-time warning SYSL
 ms.date: 10/20/2020
 ---
 
-# SYSLIB0009: The AuthenticationManager Authenticate and PreAuthenticate methods are not supported
+# SYSLIB0009: AuthenticationManager is not supported. Methods will no-op or throw PlatformNotSupportedException
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib0009.md:7]:
```suggestion
# SYSLIB0009: AuthenticationManager is not supported
```

File: docs/fundamentals/syslib-diagnostics/syslib0009.md [hunk 2]
```diff
@@ -4,13 +4,16 @@ description: Learn about the obsoletions that generate compile-time warning SYSL
...
 
 The following APIs are marked obsolete, starting in .NET 5. Use of these APIs generates warning `SYSLIB0009` at compile time and throws a <xref:System.PlatformNotSupportedException> at run time.
 
 - <xref:System.Net.AuthenticationManager.Authenticate%2A?displayProperty=nameWithType>
 - <xref:System.Net.AuthenticationManager.PreAuthenticate%2A?displayProperty=nameWithType>
 
+The entire <xref:System.Net.AuthenticationManager> class is marked as obsolete as of .NET 9. The usage of this class generates warning `SYSLIB0009` at compile time.
+The methods will no-op or throw <xref:System.PlatformNotSupportedException>.
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib0009.md:15]:
```suggestion
The methods in this class either no-op or throw a <xref:System.PlatformNotSupportedException> at run time.
```

File: docs/fundamentals/syslib-diagnostics/syslib0009.md [hunk 3]
```diff
@@ -4,13 +4,16 @@ description: Learn about the obsoletions that generate compile-time warning SYSL
...
 
-# SYSLIB0009: The AuthenticationManager Authenticate and PreAuthenticate methods are not supported
+# SYSLIB0009: AuthenticationManager is not supported. Methods will no-op or throw PlatformNotSupportedException
 
 The following APIs are marked obsolete, starting in .NET 5. Use of these APIs generates warning `SYSLIB0009` at compile time and throws a <xref:System.PlatformNotSupportedException> at run time.
 
 - <xref:System.Net.AuthenticationManager.Authenticate%2A?displayProperty=nameWithType>
 - <xref:System.Net.AuthenticationManager.PreAuthenticate%2A?displayProperty=nameWithType>
 
+The entire <xref:System.Net.AuthenticationManager> class is marked as obsolete as of .NET 9. The usage of this class generates warning `SYSLIB0009` at compile time.
```
Comment (gewarren) [inline docs/fundamentals/syslib-diagnostics/syslib0009.md:14]:
```suggestion
In .NET 9 and later versions, the entire <xref:System.Net.AuthenticationManager> class is marked obsolete. Use of this class generates warning `SYSLIB0009` at compile time.
```

**Review by gewarren — CHANGES_REQUESTED** [review]

**Review by gewarren — APPROVED** [review]

**Comment by gewarren** [comment]
@wfurt It can be merged if we change the date that's causing a warning, but why publish .NET 9 documentation now when the first preview won't be available until February?

**Comment by wfurt** [comment]
so can complete runtime issue tracking it :)

File: docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+---
+title: "Breaking change: .NET 9 obsoletions with custom IDs"
+titleSuffix: ""
+description: Learn about the .NET 9 breaking change in core .NET libraries where some APIs have been marked as obsolete with a custom diagnostic ID.
+ms.date: 10/15/2023
```
Comment (gewarren) [inline docs/core/compatibility/core-libraries/9.0/obsolete-apis-with-custom-diagnostics.md:5]:
```suggestion
ms.date: 04/05/2024
```

File: docs/core/compatibility/9.0.md [hunk 5]
```diff
@@ -58,3 +59,7 @@ If you're migrating an app to .NET 9, the breaking changes listed here might aff
 | Title                                                                                   | Type of change      | Introduced version |
 |-----------------------------------------------------------------------------------------|---------------------|--------------------|
 | [`GetXmlNamespaceMaps` type change](wpf/9.0/xml-namespace-maps.md) | Behavioral change/Source incompatible | Preview 3 |
+
+## See also
+
+- [What's new in .NET 9](../whats-new/dotnet-9.md)
```
Comment (gewarren) [inline docs/core/compatibility/9.0.md:65]:
```suggestion
- [What's new in .NET 9](../whats-new/dotnet-9/overview.md)
```

**Pull request merged**